android/src/main/java/com/rncamerakit/CKCamera.kt

```diff
@@ -307,9 +308,42 @@ class CKCamera(context: ThemedReactContext) : FrameLayout(context), LifecycleObs
         val useCases = mutableListOf(preview, imageCapture)
 
         if (scanBarcode) {
-            val analyzer = QRCodeAnalyzer { barcodes ->
-                if (barcodes.isNotEmpty()) {
-                    onBarcodeRead(barcodes)
+            val analyzer = QRCodeAnalyzer { results ->
+                if (results.isNotEmpty()) {
+                    val filteredResults = if (barcodeFrame !== null) {
+                        results.filter { (barcode, imageSize) ->
+                            val barcodeBoundingBox = barcode.boundingBox
+                            if (barcodeBoundingBox === null) {
+                                return@filter false
+                            }
```

Comment:
You can write that in one line. This will return early if variable is null

```suggestion
                            val barcodeBoundingBox = barcode.boundingBox ?: return@filter false
```